src/detect.rs

```diff
@@ -18,3 +23,196 @@ pub enum DetectOutcome {
     Pass(BuildPlan),
     Fail,
 }
+
+enum BuildpackTomlArg<BM> {
+    Path(PathBuf),
+    Str(String),
```

Comment:
> Naming: Can we call this smth. like BuildpackTomlString?

Yes, easy change.

> In general, do we need the ability to use a string as the buildpack TOML in testing? It feels like this is promoting stringly typed programming.

I agree with the hesitation to accept ANY string to ANY method. Ultimately since the string passed in is put through a toml parser it must at least be valid toml which does give us some guarantees (though not until runtime).

Accepting a string has the benefit of being able to stub out contents for tests without having to write anything to disk.

The main reason I added this feature was to make pointing at relative files easy through `include_str!` which doesn't require any dependencies. 

I use it a lot in the tests:

```rs
        let context = DetectContextBuilder::new("heroku-20")
            .buildpack_toml_from_string(include_str!(
                "../examples/example-02-ruby-sample/buildpack.toml"
            ))
            .platform(GenericPlatform::from_path(".").unwrap())
            .app_dir(".")
            .buildpack_dir(".")
            .build::<std::io::Error>()
            .unwrap();

        fake_detect(context);
```

I tried to see if maybe there are other ways to point at files relative to the current file I found https://stackoverflow.com/questions/30511331/getting-the-absolute-path-from-a-pathbuf

Playing around I also found this works:

```
        let foo = PathBuf::from_str(file!()).unwrap().parent().unwrap().join("../examples/example-02-ruby-sample/buildpack.toml");
        println!("{:?}", foo);
        println!("{:?}", foo.exists());
```

But it's...gnarly. Maybe there's a better way to point at a relative path?

I know we can also generate paths from the cargo manifest dir:

```
    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
        .join("test/fixtures")
        .join(fixture_name)
        .join("Procfile")
```

I'm personally in favor of keeping this option, I'm also happy to hear more concerns or objections.

Reply:
I've re-written the logic but retained the string option. I'm now using enums and `From` to enable only needing one entry-point.